### ISSUE TITLE
Relax domain check to allow working with Github Enterprise hosted instances

### ISSUE THREAD

**Issue by heckler** [nedbat/dinghy#3]
[Github enterprise](https://github.com/enterprise) allows organizations to host their own Github instances.  It is the same Github functionality and API, just under somecorporatedomain.com

digest.py's [method_from_url](https://github.com/nedbat/dinghy/blob/546bf5e8314728e9272d66c268eb5dd22bd06510/src/dinghy/digest.py#L156) checks the provided URL against a predefined set of URLs, but does so using the full domain (github.com).   This precludes using a URL from a private instance, such as <https://github.somecorporatedomain.com/myorg/myrepo>.

A couple thoughts on possible workarounds:

1. Just ignore the domain part of the URL on the test, assume callers are passing a URL which points to a Github instance (but then you'll need to "lift" that domain name when creating the GraphqlHelper on [line 34](https://github.com/nedbat/dinghy/blob/546bf5e8314728e9272d66c268eb5dd22bd06510/src/dinghy/digest.py#L34)
2. Add a cmdline parameter for specifying a base URL, defaulting to github.com if omitted

**Comment by nedbat** [comment]
In the back of my mind is the idea that this could eventually also pull information from non-GitHub sources, so I'd like something like this in the config file:
```
hosts:
  github.myenterprise.com: github
  api.company.com: github
```

**Comment by nedbat** [comment]
@HenryGessau suggests this as an approach:  https://github.com/nedbat/dinghy/compare/main...HenryGessau:enterprise

It implicitly uses whatever hostname is in the URLs, and assume they are all the same, but I think that is fine for a simple implementation.

**Comment by nedbat** [comment]
This is implemented in #6, and is released as part of [0.9.0](https://pypi.org/project/dinghy/0.9.0/).
